src/ics23_impl.rs

```diff
@@ -1,18 +1,19 @@
 use anyhow::{anyhow, Result};
 
-use crate::{storage::TreeReader, JellyfishMerkleTree, Version};
+use crate::{storage::TreeReader, JellyfishMerkleTree, KeyHash, SimpleHasher, Version};
 
-impl<'a, R> JellyfishMerkleTree<'a, R>
+impl<'a, R, H> JellyfishMerkleTree<'a, R, H>
```

Comment:
I'm honestly not a big fan of the way that `HashMap` handles that. Having all of your `HashMap::new()` calls fail if you try to swap out the Hasher is a pretty annoying. What if we export a type alias instead? Something like

```rust
pub type DefaultJMT<'a, R> = JellyfishMerkleTree<'a, R, Sha256>;
```

Reply:
Added a type alias `Sha2JMT`. On further reflection, I'm a little wary of adding a `DefaultJMT`.  That feels a little dangerous, since it's not clear under what circumstances that "default" hash function might change. For example, if `Sha4` comes out and has 10x the throughput of Sha2, do we still keep Sha256 in the `DefaultJMT`?

Reply:
Ah, good thinking! I'll push a quick fix

Reply:
Done!